src/tools/r2rdump/DisassemblingTypeProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Text;
+
+namespace R2RDump
+{
+    internal class DisassemblingGenericContext
```

Comment:
This was borrowed from CoreFx right?

Reply:
Yes, from corefx/src/System.Reflection.Metadata/tests/Metadata/Decoding/DisassemblingTypeProvider.cs

Reply:
@jkotas is there any sort of disclosure that needs to be included in this source file which originated from CoreFx?

Reply:
Not required.